### PR TITLE
ref(hybrid-cloud): api/endpoints/test_project_* silo coverage

### PR DESCRIPTION
**Pull request by mikejihbe**
This one is straightforward. Only minor note is that I added an `all_silo_endpoint` annotation to handle special cases like the catchall endpoint.

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
Relay-related changes LGTM.

File: src/sentry/services/hybrid_cloud/auth/__init__.py [hunk 1]
```diff
@@ -85,6 +85,12 @@ def authenticate(self, request: Request) -> Tuple[Any, Any] | None:
 
         return None
 
+    # Not sure this is technically correct, but it maintains existing behavior
+    # this is required so that we properly return 401 vs 403 status codes required
+    # by the test suite. Reasonable arguments could be made that this should be changed
+    def authenticate_header(self, request: AuthenticationRequest) -> str:
+        return 'xBasic realm="api"'
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/auth/__init__.py:92]: Should the `x` be here?

Comment (mikejihbe) [reply src/sentry/services/hybrid_cloud/auth/__init__.py:92]: Yes. Again, not sure why, but that is what the non-silo enforced test returns, so I kept it that way. I'll add an assertion to the suite.

Comment (corps) [reply src/sentry/services/hybrid_cloud/auth/__init__.py:92]: I'll take a look here -- it's like that we need to simply adjust the api endpoints hybrid cloud authentication logic.

**Comment by corps** [comment]
Closing in favor of 7

**Pull request closed without merging**